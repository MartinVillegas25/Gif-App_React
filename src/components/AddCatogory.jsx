import { useState } from "react"
import PropTypes from "prop-types"


export const AddCatogory = ( { onNewcategoty}) => {


    const [ inputValue, setInputValue] = useState()

    //vamos a usuar el metodo onChange para poder cambiar el valor del input

    const onInputChange = (event)=>{
            setInputValue(event.target.value)
    }

    const onSubmit = (e)=>{
            e.preventDefault();
            const inputValueTrim = inputValue || "";
            if(inputValueTrim.trim().length <= 1) return;
            setInputValue( "")
            onNewcategoty(inputValue.trim());
    }
  return (
    <form onSubmit={ onSubmit} aria-label="form">
        <input
            type="text"
            placeholder="Buscar Gif"
            value= {inputValue}
            onChange={onInputChange}
        />

    </form>
    
  )
}


AddCatogory.propTypes = {
  onNewcategoty: PropTypes.func.isRequired,
}