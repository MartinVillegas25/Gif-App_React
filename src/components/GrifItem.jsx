
import PropTypes from 'prop-types';



export const GrifItem = ( {title , url}) => {
  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}


GrifItem.propTypes = {
  title : PropTypes.string.isRequired,
  url : PropTypes.string.isRequired
}

