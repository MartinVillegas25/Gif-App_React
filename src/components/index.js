//archivo barril

export * from './AddCatogory'
export * from './GrifGrid'
export * from './GrifItem'