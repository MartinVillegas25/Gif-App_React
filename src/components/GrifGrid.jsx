import {PropTypes } from 'prop-types'
import { GrifItem } from "./GrifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";




export const GrifGrid = ( {category}) => {
    

//vamos a crear un custom hook para reemplazar toto el codigo abajo

const { images , isLoading } = useFetchGifs(category) // le podemos poner cualquier nombre, solo hay q poner  use delante del nombre, norma al crear hooks

  // const [images, setImages] = useState([]);

  // const getImages = async ()=>{
  //   const newImages = await getGif(category);
  //   setImages(newImages);
    
  // }

  // //el useEffect es un hook que sirva para disparar un efecto secundario, q es un algo que you quiero dispadar cuando algo sucesa

  // useEffect(() => {
  //   getImages();
    
  // }, []) 
  
  //con las llaves vacias, le digo q solo lo renderice la primera vez
  

  
    return (
    <>
        <h3>{category}</h3>

        {
          isLoading && (<h2>Loading...</h2>)
        }

        <div className="card-grid">
          {
            images.map( (img) => (
              <GrifItem 
              key={img.id}
              {...img} // para que el gifitem reciba cada una de las propediades como items
              />
            ))

          }
        </div>
        
    
    </>
  )
}

GrifGrid.propTypes = {
  category : PropTypes.string.isRequired

}
