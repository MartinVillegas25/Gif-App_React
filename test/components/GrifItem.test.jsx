import { render, screen } from "@testing-library/react";
import { GrifItem } from "../../src/components/GrifItem";

describe('pruebas en GrifItem', () => { 

    const title = 'goku';
    const url = 'https://github.com/goku'

    test('debe de hacer match con el snapshot', () => { 
         const {container} = render( <GrifItem title={title} url={url}/>) 
         expect(container).toMatchSnapshot();
        
     });

     test('debe de mostrar la imagen con la URL y el alt indicado', () => { 
        render( <GrifItem title={title} url={url}/>);
        
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
      })

      test('el titulo debe estar mostrado como texto', () => { 
        render( <GrifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();


       })

 })