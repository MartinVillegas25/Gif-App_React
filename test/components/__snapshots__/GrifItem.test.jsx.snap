// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GrifItem debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="goku"
      src="https://github.com/goku"
    />
    <p>
      goku
    </p>
  </div>
</div>
`;
